workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md

```diff
@@ -0,0 +1,527 @@
+
+```bash
+helm init --service-account tiller --wait
+```
+
+### 4. Secure Helm <a name="helm-step4"></a>
+
+Secure `tiller` from access inside the cluster.
+
+`tiller`s port is exposed in the cluster without authentication and if you probe this port _directly_ (i.e. by bypassing `helm`) then `tiller`s permissions can be exploited.
+This step forces `tiller` to listen to commands from `localhost` (i.e. `helm`) _only_ so that e.g. other pods inside the cluster cannot ask `tiller` to install a new chart granting them arbitrary, elevated RBAC privileges and exploit them.
+More details [here](https://engineering.bitnami.com/articles/helm-security.html).
+
+```bash
+kubectl patch deployment tiller-deploy \
+    --namespace=kube-system \
+    --type=json \
+    --patch='[{
+        "op": "add",
+        "path": "/spec/template/spec/containers/0/command",
+        "value": ["/tiller", "--listen=localhost:44134"]
+    }]'
+```
+
+### 5. Verify the installation <a name="helm-step5"></a>
+
+To verify the correct versions have been installed properly, run the following command.
+
+```bash
+helm version
+```
+
+You must have at least version 2.11.0 and the client (`helm`) and server (`tiller`) versions must match.
+
+Example output:
+```bash
+Client: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
+Server: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
+```
+
+## Setup BinderHub <a name="binderhub"></a>
+
+Adapted from [Zero-to-BinderHub: Setup BinderHub](https://binderhub.readthedocs.io/en/latest/setup-binderhub.html).
+
+### 1. Preparing to Install <a name="bh-step1"></a>
+
+Before we install a BinderHub, we need to configure several pieces of information and save them in `yaml` files.
+
+Create a folder named after your BinderHub.
+```bash
+mkdir shf_test_hub
+cd shf_test_hub
+```
+
+We created this folder at the same level as the cluster folder we created in [Step 4: Choose a Cluster Name](#aks-step4) (i.e. in `~/Desktop`).
+
+> **Discussion topic:** Where is a sensible place to keep _this_ folder?
+
+Create two random tokens:
+```bash
+openssl rand -hex 32
+openssl rand -hex 32
+```
+**N.B.:** The command is run **twice** as we need two different tokens.
+
+### 2. Create a `secret.yaml` file <a name="bh-step2"></a>
+ 
+Create a `secret.yaml` file containing the following config and save it in the folder we created in [Step 1: Preparing to Install](#bh-step1).
+```yaml
+jupyterhub:
+  hub:
+    services:
+      binder:
+        apiToken: "<output of FIRST 'openssl rand -hex 32' command>"
+  proxy:
+    secretToken: "<output of SECOND 'openssl rand -hex 32' command>"
+```
+
+To connect to DockerHub, add the following lines.
+```yaml
+registry:
+  username: <docker-id>
+  password: <password>
+```
+**N.B.:** `registry` is on the same level as `jupyterhub`.
```

Comment:
Maybe same _indentation_ level. In case people aren't familiar with yaml.

Reply:
Good call, I was struggling with how to word this. I conceptually understood what I meant, but words failed me a touch.